library/src/main/java/org/xmtp/android/library/Conversation.kt

```diff
@@ -28,154 +29,130 @@ sealed class Conversation {
     data class V2(val conversationV2: ConversationV2) : Conversation()
 
     data class Group(val group: org.xmtp.android.library.Group) : Conversation()
+    data class Dm(val dm: org.xmtp.android.library.Dm) : Conversation()
 
-    enum class Version { V1, V2, GROUP }
+    enum class Version { V1, V2, GROUP, DM }
 
-    // This indicates whether this a v1 or v2 conversation.
     val version: Version
         get() {
             return when (this) {
                 is V1 -> Version.V1
                 is V2 -> Version.V2
                 is Group -> Version.GROUP
+                is Dm -> Version.DM
+            }
+        }
+
+    val id: String
+        get() {
+            return when (this) {
+                is V1 -> throw XMTPException("Only supported for V3")
+                is V2 -> throw XMTPException("Only supported for V3")
+                is Group -> group.id
+                is Dm -> dm.id
+            }
+        }
+
+    val topic: String
+        get() {
+            return when (this) {
+                is V1 -> conversationV1.topic.description
+                is V2 -> conversationV2.topic
+                is Group -> group.topic
+                is Dm -> dm.topic
             }
         }
 
-    // When the conversation was first created.
     val createdAt: Date
         get() {
             return when (this) {
                 is V1 -> conversationV1.sentAt
                 is V2 -> conversationV2.createdAt
                 is Group -> group.createdAt
+                is Dm -> dm.createdAt
             }
         }
 
-    // This is the address of the peer that I am talking to.
-    val peerAddress: String
+    val name: String
```

Comment:
Add all the common interface functionality between groups and dms for ease of access.